src/wp-includes/theme.php

```diff
@@ -866,6 +866,13 @@ function switch_theme( $stylesheet ) {
 
 	update_option( 'theme_switched', $old_theme->get_stylesheet() );
 
+	// Set autoload=no for the previous theme
+	$old_theme_stylesheet = $old_theme->get_stylesheet();
+	wp_set_option_autoload( "theme_mods_$old_theme_stylesheet", 'no' );
```

Comment:
We need to also cover a potential parent theme here (i.e. if the old `template` is different from the old `stylesheet`). We could then use `wp_set_options_autoload()` to reduce the database requests needed.
```suggestion
	$old_theme_mods = array_unique(
		array(
			"theme_mods_{$old_theme->get_stylesheet()}",
			"theme_mods_{$old_theme->get_template()}"
		)
	);
	wp_set_options_autoload( $old_theme_mods, 'no' );
```